interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java

```diff
@@ -351,23 +353,25 @@ protected ManagedChannel createChannel() {
                 ? TestUtils.newSslSocketFactoryForCa(Platform.get().getProvider(),
                     TestUtils.loadCert("ca.pem"))
                 : (SSLSocketFactory) SSLSocketFactory.getDefault();
-            builder.sslSocketFactory(factory);
+            okBuilder.sslSocketFactory(factory);
           } catch (Exception e) {
             throw new RuntimeException(e);
           }
         } else {
-          builder.usePlaintext(true);
+          okBuilder.usePlaintext(true);
         }
         if (fullStreamDecompression) {
-          builder.enableFullStreamDecompression();
+          okBuilder.enableFullStreamDecompression();
         }
-        return builder.build();
+        builder = okBuilder;
       }
+      io.grpc.internal.TestingAccessor.setStatsContextFactory(builder, getClientStatsFactory());
+      return builder.build();
     }
 
     @Override
-    protected boolean metricsExpected() {
```

Comment:
It looks like metricsExpected() will now return true for the interop client - only serverInProcess() returning false is required now?

Reply:
Yes. serverInProcess() is the most precise control.